File: Java/Program/SmartSpeaker.java
// Interface untuk SmartDevice karena Java tidak mendukung multiple inheritance class
interface SmartDeviceInterface {
    void setWifiEnabled(boolean wifiEnabled);
    void setNetworkName(String networkName);
    boolean getWifiEnabled();
    String getNetworkName();
}

public class SmartSpeaker extends Speaker implements SmartDeviceInterface {
    private SmartDevice smartDevice;
    private String asisten;

    // Constructor
    public SmartSpeaker() {
        super();
        this.smartDevice = new SmartDevice();
        this.asisten = "";
    }

    public SmartSpeaker(String merk, String nama, int harga, String jenisKoneksi, boolean terhubung,
                       int dayaOutput, boolean stereo, boolean wifiEnabled, String networkName,
                       String asisten) {
        super(merk, nama, harga, jenisKoneksi, terhubung, dayaOutput, stereo);
        this.smartDevice = new SmartDevice(wifiEnabled, networkName);
        this.asisten = asisten;
    }

    // Setter methods
    public void setAsisten(String asisten) { this.asisten = asisten; }
    public void setWifiEnabled(boolean wifiEnabled) { smartDevice.setWifiEnabled(wifiEnabled); }
    public void setNetworkName(String networkName) { smartDevice.setNetworkName(networkName); }

    // Getter methods
    public String getAsisten() { return asisten; }
    public boolean getWifiEnabled() { return smartDevice.getWifiEnabled(); }
    public String getNetworkName() { return smartDevice.getNetworkName(); }

    // Tampilkan Info
    @Override
    public String tampilkanInfo() {
        return super.tampilkanInfo() + " | " + smartDevice.tampilkanInfoSmart() +
               " | Asisten: " + (asisten.isEmpty() ? "Tidak ada" : asisten);
    }
}

File: Python/Program/SmartSpeaker.py
from Speaker import Speaker
from SmartDevice import SmartDevice

class SmartSpeaker(Speaker, SmartDevice):
    def __init__(self, merk="", nama="", harga=0, jenis_koneksi="", terhubung=False, daya_output=0, stereo=True, wifi_enabled=False, network_name="", asisten=""):
        Speaker.__init__(self, merk, nama, harga, jenis_koneksi, terhubung, daya_output, stereo)
        SmartDevice.__init__(self, wifi_enabled, network_name)
        self.asisten = asisten
    
    # Setter
    def set_asisten(self, asisten):
        self.asisten = asisten
    
    # Getter
    def get_asisten(self):
        return self.asisten
    
    # Tampilkan Info
    def tampilkan_info(self):
        speaker_info = Speaker.tampilkan_info(self)
        smart_info = SmartDevice.tampilkan_info_smart(self)
        asisten = self.asisten if self.asisten else "Tidak ada"
        return f"{speaker_info} | {smart_info} | Asisten: {asisten}"